### ISSUE TITLE
Switch to native Promises in framework code

### ISSUE THREAD

**Issue by cboulanger** [qooxdoo/qooxdoo#9877]
Given universal support of Promises in Browsers and Node (and the availability of polyfills), there isn't a reason to provide our own Promise implementation (`qx.Promise`) any more once qooxdoo switches to support ECMAScript 6, since it introduces backwards-compatibility issues that we don't want to have to deal with, and might also be a performance issue.

**Comment by johnspackman** [comment]
i dont think we should deprecate qx.Promise - it adds a load of features that are useful (eg static methods such as `.all` etc as well as member methods) and which can be reimplemented on top of native Promise; however, ditching bluebird seems reasonable if all supported browsers provide native promises.

**Comment by goldim** [comment]
Tried to cut off blubird promises and use only native promises instead. But got senior trouble:
Native promise can't be cancellable. There is some work around but not sure about it.

**Comment by johnspackman** [comment]
I think it's possible, eg I've just read this: https://medium.com/@masnun/creating-cancellable-promises-33bf4b9da39c

**Comment by cboulanger** [comment]
Let's try to do this for qx. 8.0 - there is no need for an underlying Bluebird implementation in 2025...

**Comment by cboulanger** [comment]
@johnspackman I just looked at the [source for qx.Promise](https://github.com/qooxdoo/qooxdoo/blob/master/source/class/qx/Promise.js) and it looks like you already set it up so that a different implementation could be used - however, there is no API for that, is there? Do you have any clue where the ["unhandled promise rejection" error](https://github.com/qooxdoo/qooxdoo/issues/10739) could originate from that [only shows up in Firefox](https://github.com/qooxdoo/qooxdoo/pull/10742#issuecomment-2566984707)?

**Comment by cboulanger** [comment]
As you know the code best, do you think we could have a BC fix that would allow me to use a native promise instead of Bluebird when creating a new `qx.Promise` instance? Could Bluebird be prevented from overriding the `Promise` symbol?

**Comment by cboulanger** [comment]
It seems like a lot of the binding stuff has become obsolete since we have `() => {}` - maybe the wrapping could be omitted for when the promise resolver is a lamda?

**Comment by johnspackman** [comment]
> As you know the code best, do you think we could have a BC fix that would allow me to use a native promise instead of Bluebird when creating a new `qx.Promise` instance?

Yes, but it needs some work - there are a number of additional methods (eg `qx.Promise.all`) which are implemented by Bluebird - so in order to remove Bluebird, we need to implement `.all` etc.  It should be fairly straightforward to do, but will require some legwork to add unit tests etc

> It seems like a lot of the binding stuff has become obsolete since we have () => {} - maybe the wrapping could be omitted for when the promise resolver is a lamda?

Personally, i'd say if we were starting again today, then yes ... but it would be a breaking change.  We could do something where it's detected and a deprecated warning is output to the console?  And get all of this into 7.x, so that 8.0 removes support for it?

**Comment by johnspackman** [comment]
> Do you have any clue why the ["unhandled promise rejection" error](https://github.com/qooxdoo/qooxdoo/issues/10739) [only shows up in Firefox](https://github.com/qooxdoo/qooxdoo/pull/10742#issuecomment-2566984707)?

No, but if you can get me a reproducable example Ill take a look

**Comment by goldim** [comment]
@johnspackman 
```javascript
(async () => {
try {
const promise = new qx.Promise((resolve, reject) => {
reject(new Error("Rejecting first promise"));
});
await new qx.Promise((resolve) => setTimeout(() => {
console.log("Resolving second promise")
resolve();
}, 1000));
await promise;
} catch (reason) {
console.log("Caught the first promise rejection");
}
})();
```

**Comment by cboulanger** [comment]
See also https://github.com/qooxdoo/qooxdoo/issues/10739#issuecomment-2568311031

**Comment by johnspackman** [comment]
> It was written by @derrell and shared in gitter.

Darn.  Gitter is broken for me.  Authentication just will not work since I switched laptop.  Ill try again with it

**Comment by goldim** [comment]
@p9malino26 Hello, you could find my work useful if you gonna. I was stuck with some unit tests. My branch is https://github.com/goldim/qooxdoo/pull/2 . You may take my changes.

**Comment by goldim** [comment]
Last times I've started to keep the philosophy "Don't touch it if works".  Specifically about promises the little "downside" of `qx.Promise` is extra dependency and extra weight of it in qooxdoo. But it is not crusial imho.

**Comment by johnspackman** [comment]
Closing this because it's been implemented
